Src/FluentAssertions/Equivalency/ConstraintCollectionEquivalencyStep.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Data;
+using System.Linq;
+
+using FluentAssertions.Execution;
+
+namespace FluentAssertions.Equivalency
+{
+    public class ConstraintCollectionEquivalencyStep : IEquivalencyStep
+    {
+        public bool CanHandle(IEquivalencyValidationContext context, IEquivalencyAssertionOptions config)
+        {
+            return typeof(ConstraintCollection).IsAssignableFrom(config.GetExpectationType(context));
+        }
+
+        public bool Handle(IEquivalencyValidationContext context, IEquivalencyValidator parent, IEquivalencyAssertionOptions config)
+        {
+            var subject = (ConstraintCollection)context.Subject;
```

Comment:
🤔 You know that the `expectation` is of that because of the `CanHandle` method, but you can't make that assumption about the `subject`. Same for `ConstraintEquivalencyStep `

Reply:
That's fair. I feel like it's fair to assume, though, that the subject won't be `null` for `Constraint`, `ContraintCollection` and `DataRowCollection`, since they'll only be arrived at through walking the tree of a `DataTable`. There's no top-end `.Should(this DataRowCollection)` for instance. Does that seem a reasonable assumption to you?

Reply:
Sure.